src/platform/packages/shared/kbn-workflows/common/utils/find_inputs_in_graph/find_inputs_in_graph.ts

```diff
@@ -44,17 +56,12 @@ export function findInputsInGraph(workflowGraph: WorkflowGraph): Record<string,
         stepInputs.push((node as EnterForeachNode).configuration.foreach);
         stepInputsKey = enterForeachNode.stepId;
       }
-    }
-    if ((node as AtomicGraphNode).type === 'atomic') {
-      const atomicNode = node as AtomicGraphNode;
-      stepInputsKey = atomicNode.stepId;
-      Object.values(atomicNode.configuration.with).forEach((input) => {
-        if (typeof input !== 'string') {
-          return;
-        }
-
-        extractTemplateVariables(input).forEach((variable) => stepInputs.push(variable));
-      });
+    } else {
+      // We try to scan the whole node, because otherwise, we would need a special case for each node type such as http, kibana, elasticsearch, etc
+      // Not good, most likely and other nodes will need to be subset of atomic node, or something else
```

Comment:
This comment is mainly relevant for understanding the differences from the existing code during the review process. Once the PR is merged, imho it will no longer be relevant

I’d suggest either removing it or rewriting it to be more concise

Reply:
With that comment I try to explain why the approach of scanning node is used.
I think it's worth having it until we make http/elasticsearch/kibana steps being a subset of AtomicNode.